### ISSUE TITLE
Fails to build on GCC 10, 11, and 12, "the address of ‘glz::opts{10, false}’ is not a valid template argument"

### ISSUE THREAD

**Issue by Ahajha** [stephenberry/glaze#4]
include/glaze/core/write.hpp:24:58: error: the address of ‘glz::opts{10, false}’ is not a valid template argument
   24 |             detail::write<Opts.format>::template op<Opts>(std::forward<T>(value), buffer, ix);

With basic test program:
```c++
#include <glaze/json/write.hpp>

#include <iostream>
#include <string>

struct my_struct {
  int i = 287;
  double d = 3.14;
  std::string hello = "Hello World";
  std::array<uint64_t, 3> arr = {1, 2, 3};
};

template <> struct glz::meta<my_struct> {
  using T = my_struct;
  static constexpr auto value = object("i", &T::i,         //
                                       "d", &T::d,         //
                                       "hello", &T::hello, //
                                       "arr", &T::arr      //
  );
};

int main() {
  my_struct s;
  std::string buffer;
  glz::write_json(s, buffer);
  std::cout << buffer << '\n';
}
```

**Comment by stephenberry** [comment]
Thanks, I've just been testing with clang and MSVC, but want to support gcc. Will look into this.

**Comment by stephenberry** [comment]
I fixed the error you reported above, but it looks like gcc has some issues with std::declval, which I'm not sure how to get around.
Specifically this line `return std::declval<T>().*std::declval<mptr_type>();` in member_check in common.hpp. gcc seems to think this std::declval is being evaluated, even when I make the function consteval. Until gcc fixes the bug or I figure out a work around gcc just won't work yet. clang and MSVC should be working fine.

**Comment by stephenberry** [comment]
I'm closing this issue because the problem addressed above has been fixed, but I added a new issue #7 for gcc std::declval issues and why it doesn't currently build.
